gpytorch/models/exact_gp.py

```diff
@@ -61,6 +61,57 @@ def set_train_data(self, inputs=None, targets=None, strict=True):
             self.train_targets = targets
         self.prediction_strategy = None
 
+    def get_fantasy_model(self, inputs, targets, **kwargs):
+        if self.prediction_strategy is None:
+            raise RuntimeError("Fantasy observations can only be added after making predictions with a model so that "
```

Comment:
No because creating the fantasy prediction strategy lives inside of prediction strategy, which we don't have or know what it should be without calling the model in eval mode first.